### PR TITLE
use method decorator to handle common UCR exceptions

### PR DESCRIPTION
**Pull request by nickpell**
avoids duplicating try-except logic 3 times

### CODE REVIEW

File: corehq/apps/userreports/decorators.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+)
+from corehq.util.soft_assert import soft_assert
+
+_soft_assert = soft_assert(
+    to='{}@{}'.format('npellegrino+ucr-get-data', 'dimagi.com'),
+    exponential_backoff=False,
+)
+
+
+def catch_and_raise_exceptions(func):
```
Comment (nickpell) [inline corehq/apps/userreports/decorators.py]: open to naming suggestions

**Comment by nickpell** [comment]
@NoahCarnahan @czue

File: corehq/apps/userreports/decorators.py (hunk 1 above)
Comment (NoahCarnahan) [inline corehq/apps/userreports/decorators.py]: `catch_and_raise_ucr_exceptions()` maybe? I don't feel strongly.

**Comment by NoahCarnahan** [comment]
looks good to me, build failure seems unrelated.

**Comment by nickpell** [comment]
kicked build

**Comment by nickpell** [comment]
tests are still failing after kicking build for second time.  looks unrelated, but wondering if there's something wrong with tests

**Comment by NoahCarnahan** [comment]
This seems to be the error:
```
Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "/usr/local/lib/python2.7/logging/__init__.py", line 1669, in shutdown
    h.release()
  File "/vendor/lib/python2.7/site-packages/cloghandler.py", line 194, in release
    self.stream.flush()
ValueError: I/O operation on closed file
```
which, based on the slack discussion this morning, sounds like an issue with shut down of management commands, so I think all the tests ran, but this came up during shutdown. So I'm going to merge.

**Comment by czue** [comment]
This PR looks like it was merged incorrectly. It also still causes a test failure even with the proper merge due to the wrong import being used. https://github.com/dimagi/commcare-hq/pull/10899

**Pull request merged**